Tests/SonarScanner.MSBuild.PreProcessor.Test/RoslynAnalyzerProviderTests.cs

```diff
@@ -78,21 +82,158 @@ public void RoslynConfig_ValidProfile()
     {
         var sonarProperties = new ListPropertiesProvider(new Dictionary<string, string>
         {
-            // for ruleset
+            {"wintellect.pluginKey", "wintellect"},
+            {"wintellect.pluginVersion", "1.13.0"},
+            {"wintellect.staticResourceName", "wintellect.zip"},
+            {"sonar.cs.analyzer.dotnet.pluginKey", "csharp"},
+            {"sonar.cs.analyzer.dotnet.pluginVersion", "1.42.0"},
+            {"sonar.cs.analyzer.dotnet.staticResourceName", "SonarAnalyzer.zip"},
+            {"sonar.vb.testPropertyPattern", "foo"},
+            {"sonar.cs.testPropertyPattern", "foo"},
+            {"sonar.sources", "**/*.*"},
+            {"sonar.cs.foo", "bar"}
+        });
+        var context = new Context(TestContext, sonarProperties, [[@"c:\assembly1.dll"], [@"d:\foo\assembly2.dll"]]);
+
+        var expectedSonarLintXml = """
+            <?xml version="1.0" encoding="UTF-8"?>
+            <AnalysisInput>
+              <Settings>
+                <Setting>
+                  <Key>sonar.cs.analyzer.dotnet.pluginKey</Key>
+                  <Value>csharp</Value>
+                </Setting>
+                <Setting>
+                  <Key>sonar.cs.analyzer.dotnet.pluginVersion</Key>
+                  <Value>1.42.0</Value>
+                </Setting>
+                <Setting>
+                  <Key>sonar.cs.analyzer.dotnet.staticResourceName</Key>
+                  <Value>SonarAnalyzer.zip</Value>
+                </Setting>
+                <Setting>
+                  <Key>sonar.cs.testPropertyPattern</Key>
+                  <Value>foo</Value>
+                </Setting>
+                <Setting>
+                  <Key>sonar.cs.foo</Key>
+                  <Value>bar</Value>
+                </Setting>
+              </Settings>
+              <Rules>
+                <Rule>
+                  <Key>S1116</Key>
+                  <Parameters>
+                    <Parameter>
+                      <Key>key</Key>
+                      <Value>value</Value>
+                    </Parameter>
+                  </Parameters>
+                </Rule>
+                <Rule>
+                  <Key>S1125</Key>
+                </Rule>
+              </Rules>
+              <Files>
+              </Files>
+            </AnalysisInput>
+
+            """;
+
+        context.AssertCorrectAnalyzerSettings();
+        context.AssertNoWarningsOrErrors();
+        context.AssertCorrectRulesets();
+        context.AssertExpectedAssemblies(@"c:\assembly1.dll", @"d:\foo\assembly2.dll");
+        context.AssertExpectedPluginsRequested(
+            [new() { Key = "wintellect", Version = "1.13.0", StaticResourceName = "wintellect.zip" },
+            new() { Key = "csharp", Version = "1.42.0", StaticResourceName = "SonarAnalyzer.zip" }]);
+        context.AssertExpectedAdditionalFileExists(expectedSonarLintXml);
+    }
+
+    [TestMethod]
+    public void RoslynConfig_ValidProfile_Legacy_Only()
```

Comment:
We use `_` to separate logical groups, not individual words
```suggestion
    public void RoslynConfig_ValidProfile_LegacyOnly()
```